tests/blockchain/Test_BlockchainAPI.cpp

```diff
@@ -857,6 +858,202 @@ TEST_F(Test_BlockchainAPI, testBip44_eth_testnet)
     EXPECT_EQ(address, "47fd2e4d1887f2a503e6a439139aee378a4788e1");
 }
 
+TEST_F(Test_BlockchainAPI, testBip44_eth_testnet_example_2)
+{
+    const int bip44_purpose = 44;
+    const opentxs::Bip44Type bip44_coin_type = api_.Crypto().Blockchain().Bip44(
+        opentxs::blockchain::Type::Ethereum_ropsten);
+    const int bip44_account = 0;
+    const int bip44_change = 0;
+    const int bip44_address_index = 0;
+    auto reason = api_.Factory().PasswordPrompt("Preparing ETH crypto-context");
+    const auto bip32_path = opentxs::crypto::Bip32::Path{
+        static_cast<opentxs::Bip32Index>(bip44_purpose) |
+            static_cast<opentxs::Bip32Index>(opentxs::Bip32Child::HARDENED),
+        static_cast<opentxs::Bip32Index>(bip44_coin_type) |
+            static_cast<opentxs::Bip32Index>(opentxs::Bip32Child::HARDENED),
+        static_cast<opentxs::Bip32Index>(bip44_account) |
+            static_cast<opentxs::Bip32Index>(opentxs::Bip32Child::HARDENED),
+        static_cast<opentxs::Bip32Index>(bip44_change),
+        static_cast<opentxs::Bip32Index>(bip44_address_index),
+    };
+
+    auto seed_id = api_.InternalClient().Exec().Wallet_ImportSeed(
+        "battle drastic essay photo equip reunion entry ready boring drive "
+        "misery code",
+        "");
+
+    auto eth_key = api_.Crypto().Seed().GetHDKey(
+        seed_id,
+        opentxs::crypto::EcdsaCurve::secp256k1,
+        bip32_path,
+        opentxs::crypto::key::asymmetric::Role::Sign,
+        reason);
+
+    auto pub_key = eth_key->PublicKey();
+    auto pub_key_data = api_.Factory().DataFromBytes(pub_key);
+    auto address = api_.Crypto().Blockchain().CalculateAddress(
+        opentxs::blockchain::Type::Ethereum_ropsten,
+        opentxs::blockchain::crypto::AddressStyle::ChecksummedHex,
+        pub_key_data);
+
+    EXPECT_EQ(address, "0abd1bd2e76ca0fe98ca1b19d2980e25e8d3397e");
```

Comment:
shouldn't `CalculateAddress` return `0x0abd1bd2e76ca0fe98ca1b19d2980e25e8d3397e` already? dunno which version is more convenient for later use ;)

Reply:
Yeah, to be honest, I also ask the same question myself. For time being it was easier for me to return the address without 0x, but from the user's perspective, it would be more intuitive when we return the address with 0x.